hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQLDialect.java

```diff
@@ -691,6 +691,11 @@ public String translateExtractField(TemporalUnit unit) {
 		}
 	}
 
+	@Override
+	public String formatBinaryliteral(byte[] bytes) {
+		return "bytea '\\x" + StandardBasicTypes.BINARY.toString( bytes ) + "'";
```

Comment:
I *think* this is the best thing to do on PostgreSQL, though I'm not 100% certain.